### PR TITLE
Add nekojb to jailbreakFiles

### PR DESCRIPTION
**Pull request by ssalggnikool**
## Why?
This jailbreak has been confirmed by many (including me) to be an actual real jailbreak, and is commonly misinformed that is "fake" due to extremely early builds being judged. I don't think it's fair to keep censoring it, evelyn themselves even apologized for it being fake because it never was / supposed to be.

Apology

Screenshot of evelyn counting on hhls for the jb

Also, mineek is one of the developers of nekojb, if you still have doubts you can ask him directly.

-----
Also I don't understand appledb at its entirety all that well, so please ask me to make any changes if necessary.

### CODE REVIEW

**Comment by ssalggnikool** [comment]
Also not gonna exclude this: yes it did have licensing problems and things that were problematic in the first *public* build, but that specific build was taken down and a new build that fixed the issues was released shortly after.

And you have to consider that palera1n also had some licensing issues that were later "addressed" with the following commit [here](https://github.com/palera1n/jbinit/commit/7798b44a4bd72eb8e6d523ad094263aa99dac5fd), which is not present in any public betas in releases.

**Comment by ssalggnikool** [comment]
As I said in a comment, I mentioned that other jailbreaks can be violating licenses too, which I do feel this is just being cherry picked because of how "untrustworthy" the dev seems to be (palera1n still has legal issues with plutil) including unc0ver with their own issues. 

> the tool honestly seems to want to go in a really jank direction (think: rootful, rootless Cydia, etc.)
It shouldn't matter if it's going in a jank direction, rather if the developer wants to do it they can with their own jailbreak. You have to consider that all these "shiny" tools utilizing KFD are still allowed in rjb even though they're promoting their own package scheme and weird "semi-jailbreak" infrastructure. 

The main concern here that I have currently is the way r/Jailbreak is promoting nekojb as a "fake jailbreak", which by any means isn't true in the slightest and it drives away any new developer who's actually willing to learn and make a new jailbreak tool for people to use.

tldr; Nekojb and roothide shouldn't be filtered, at most people should instead be guided in the right direction into using "recommended" tools.

**Comment by ssalggnikool** [comment]
Also, I don't want to do anything with r/Jailbreak either but I do contribute to other tools that are used commonly throughout, so this point is highly just another cherry problem that was made up to not allow nekojb.

> At this point in time, this will not be merged - this could be reconsidered once it’s fully released and OSS’d (since hahalosah and various people around them claim that they will at some point, but have yet to do so) but as of now, this decision holds.

It being OSS should not matter in the slightest, there are obvious private techniques being used that the developer doesn't want to share publicly at this point in time, including dora and their meowbrek2 jailbreak. 

I would recommend you look at this without thinking of all the bogus requirements for you to take to reconsider.

**Comment by TheMasterOfMike** [comment]
Let's go down what you said, point by point:
> As I said in a comment, I mentioned that other jailbreaks can be violating licenses too, which I do feel this is just being cherry picked because of how "untrustworthy" the dev seems to be (palera1n still has legal issues with plutil) including unc0ver with their own issues.

There's a difference between a previous vetted tool being discovered to have license violations and a new tool having license issues (how would you possibly justify filtering or removing something like palera1n? You probably can't.).

I'm not saying this to excuse or absolve palera1n's issues, but what I'm saying is both that a trusted tool is generally put under less scrutiny (this is a fact) as well as that if a trusted tool is discovered to have license violations, it's generally:
- less significant
- less noticeable
- resolved as best as reasonably possible

palera1n isn't the only tool which has these issues - there's stuff like Sam's Substitute, which has yet to be updated (last I checked) on the repository, and is probably violating licenses as a result - but isn't significant or feasible enough to actually warrant it being filtered or pushed for its removal from rootful palera1n.

The difference with nekojb, is that not only was it a new tool (which warrants more scrutiny than an already released tool like palera1n) but also, its first public beta was extremely blatant with regards to license violations that it warrant even more scrutiny for it to ever be added and reconsidered in the future.

There is more to this, but I'll save it for the end.

> It shouldn't matter if it's going in a jank direction, rather if the developer wants to do it they can with their own jailbreak. You have to consider that all these "shiny" tools utilizing KFD are still allowed in rjb even though they're promoting their own package scheme and weird "semi-jailbreak" infrastructure.

To be clear (and this is on me for not clarifying this, I probably shouldn't have made this point in the first place) - it being jank isn't a direct factor, all it merely is is a limited contributing factor to the view of this tool as a whole, there's a reason RootHide's filter was later removed and that the semi-jailbreak stuff is allowed in r/jailbreak. It's not something that many people would recommend (or something that most can assist with) but is something that isn't outright filtered.

(As for why RootHide and the semi-jailbreak stuff aren't on AppleDB, the former is because it's really just Dopamine but patched to do its RootHide thing, and as for the latter it's not really a true jailbreak (though the latter could change in the future))

> Also, I don't want to do anything with r/Jailbreak either but I do contribute to other tools that are used commonly throughout, so this point is highly just another cherry picked problem that was made up to not allow nekojb.

I feel as though it's worth sharing context of this - there were people trying to spam nekojb Discord invites and tool url's in r/jailbreak, when I mentioned this to hahalosah (since I do have contact with them) - they mention that they:
- Didn't ask for any of this
- Don't even want to be associated with r/jailbreak

Given that their not wanting to be associated ties to both of those things (and that there were people banned from hahalosah's Discord for sharing both of these), I think one can reasonably come to the conclusion that hahalosah doesn't want this tool to be shared around in r/jailbreak anyways... and with it, adding to AppleDB would lead to its unfiltering in r/jailbreak (since why would it be added here but left filtered in r/jailbreak?), which would lead to this tool being shared around in r/jailbreak anyways (which goes entirely against what hahalosah has said).

> It being OSS should not matter in the slightest, there are obvious private techniques being used that the developer doesn't want to share publicly at this point in time, including dora and their meowbrek2 jailbreak.

The reason I mention the OSS bit has to do with two licenses that are present withing nekojb, that being AGPL and GPL.

Depending on how nekojb is using code that is based around either of those licenses, nekojb has to be open source or it's violating those licenses. While hahalosah has said that they aren't using it directly within nekojb (of which I am inclined to personally believe by the way) the past license violations combined with the uncertainty of the tool as a whole lead to there being question marks about this until it is OSS'd, at which point, it would both:
- make it reasonable to identify what is being used that's based on AGPL and GPL
- resolve the issue anyways (since being OSS would solve the issue that it... needs to be OSS...)

> I would recommend you look at this without thinking of all the bogus requirements for you to take to reconsider.

I (and many others) don't think what has been said is bogus - everything that has been said as to why it won't be added is based on a reasonable interpretation of the timeline and the facts as I (and most other people) are currently am aware of.
